lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
> don't know what's the impact when can't get IP

The request won't be handled. An app error will occur I think.
If we're not throwing, what do you think the default should be @dead-horse? Undefined? An empty string?